Add JavaScript minification, release as version 0.8.0

patrickrobrecht opened the pull request:
- Add code style check and minification for the JavaScript file (added by #14)
- Update the dependencies for the existing PHP code style check
- Prepare release of version 0.8.0 (increase version number and WP compatibility, add changelog) which fixes #10
- Add PHP 7.4 to the Travis build
- Adjust the settings for the automatic release in the SVN repository (by adding a `.distignore` as learned during the last Statify release)

patrickrobrecht committed:
Add code style check and minification for JavaScript file, update dependencies, prepare release of version 0.8.0

patrickrobrecht committed:
Add npm install to Travis build

patrickrobrecht committed:
Don't check minified JavaScript files

stklcode reviewed (changes requested):
Isn't a composer step missing in the deployment pipeline to push the minified script?

Also the plugin version in _spcl.php_ file header is still 0.7.5

patrickrobrecht committed:
Add build step to plugin deploy: composer install triggers composer build

patrickrobrecht commented:
> Isn't a composer step missing in the deployment pipeline to push the minified script?

You're right, fixed with the recent commit.

> Also the plugin version in _spcl.php_ file header is still 0.7.5

Also fixed.

florianbrinkmann reviewed (approved):
Cannot say much to the GitHub action, but the rest looks good to me. Tested the build steps locally and it works. Thanks for putting this together, @patrickrobrecht!

stklcode commented on .eslintrc.json:
```diff
@@ -0,0 +1,15 @@
+  },
+  "env": {
+    "browser": true,
+    "jquery": true
```
JQuery is superfluous here, but at least not wrong for the admin backend.

stklcode commented on composer.json:
```diff
@@ -7,17 +7,29 @@
     "cs": [
-      "phpcs --standard=phpcs.xml -s"
+      "phpcs --standard=phpcs.xml -s",
+      "npx eslint assets/js/notice.js"
     ],
     "csfix": [
-      "phpcbf --standard=phpcs.xml"
+      "phpcbf --standard=phpcs.xml",
+      "npx eslint assets/js/notice.js --fix"
```
I'd personally have split the checks for PHP and JS (considering Composer-only environments) with wrapper steps, although it's only a single line here.

patrickrobrecht replied:
That's an argument, for changes that don't affect the JavaScript I should be valid to ignore the JavaScript code style check in the development environment). So I added separate commands for the checks as well.

stklcode commented on phpcs.xml:
```diff
@@ -22,8 +22,7 @@
     <config name="minimum_supported_wp_version" value="3.7" />
-    <rule ref="WordPress">
-    </rule>
+    <rule ref="WordPress"></rule>
```
could have been self-closing

stklcode reviewed (approved):
Left some minor (pedantic) notes on the non-code changes.

Nevertheless the changes look reasonable to me and the plugin works fine after simulated deployment, so feel free to ignore :wink:

patrickrobrecht committed:
Provide own commands for code style checks
